bin/get-cutouts

```
@@ -0,0 +1,386 @@
+                outprefix, _metadata['SURVEY'], _metadata['PROGRAM'],
+                _metadata['HEALPIX'], _metadata['TARGETID']))
+        elif coadd_type == 'cumulative':
+            jpegfile = os.path.join(htmldir, 'tmp.{}-{}-{}-{}.jpeg'.format(
+                outprefix, _metadata['TILEID'], coadd_type, _metadata['TARGETID']))
+        elif coadd_type == 'pernight':
+            jpegfile = os.path.join(htmldir, 'tmp.{}-{}-{}-{}.jpeg'.format(
+                outprefix, _metadata['TILEID'], _metadata['NIGHT'], _metadata['TARGETID']))
+        elif coadd_type == 'perexp':
+            jpegfile = os.path.join(htmldir, 'tmp.{}-{}-{}-{}-{}.jpeg'.format(
+                outprefix, _metadata['TILEID'], _metadata['NIGHT'],
+                _metadata['EXPID'], _metadata['TARGETID']))
+        elif coadd_type == 'custom':
+            jpegfile = os.path.join(htmldir, 'tmp.{}-{}-{}-{}-{}.jpeg'.format(
+                outprefix, _metadata['SURVEY'], _metadata['PROGRAM'],
+                _metadata['HEALPIX'], _metadata['TARGETID']))
+        else:
+            errmsg = 'Unrecognized coadd_type {}!'.format(coadd_type)
+            print(errmsg)
+            raise ValueError(errmsg)
+        return jpegfile
+
+    if type(metadata) is astropy.table.row.Row:
+        jpegfile = _one_filename(metadata)
+    else:
+        jpegfile = [_one_filename(_metadata) for _metadata in metadata]
+    
+    return np.array(jpegfile)
+
+def plan(comm=None, specprod=None, coadd_type='healpix',
+         survey=None, program=None, healpix=None, tile=None, night=None, 
+         outdir_data='.', outdir_html='.', mp=1, overwrite=False):
+
+    from astropy.table import Table
+                                
+    t0 = time.time()
+    if comm is None:
+        rank, size = 0, 1
+    else:
+        rank, size = comm.rank, comm.size
+
+    desi_root = '/global/cfs/cdirs/desi'
+    #desi_root = os.environ.get('DESI_ROOT', DESI_ROOT_NERSC)
+    # look for data in the standard location
```

Pedantic, but... it would be preferable to do something like
```python
if 'DESI_ROOT' in os.environ:
    desi_root = os.environ['DESI_ROOT']
else:
    desi_root = '/global/cfs/cdirs/desi'
```
or equivalently but arguably more cryptically:
```python
desi_root = os.getenv('DESI_ROOT', default='/global/cfs/cdirs/desi')
```

---

I used to use `os.getenv` but somewhere, sometime, someone yelled at me to switch to `os.environ.get` and so I did. I don't think there's a functional difference between these methods and indeed according to this Stackoverflow thread, `os.getenv` is just a wrapper on `os.environ.get` (but I didn't dig to see if it's still true)--
  https://stackoverflow.com/questions/16924471/difference-between-os-getenv-and-os-environ-get

---

I think the previous issue was that `os.getenv('BLAH')` will silently return `None` if `$BLAH` isn't set, while `os.environ['BLAH']` will raise a `KeyError` so that you immediately know that the environment you were trying to get isn't set and it will fail there in an obvious way, instead of propagating a `None` through the code until that causes it to crash in some harder to debug downstream case.

In this case, it would be fine to use either `os.environ` or `os.getenv`, but the main point of the comment is to use that if the environment variable is defined and only fall back to a hardcoded path if not.

This isn't any worse than the other hardcoded paths I commented on; it's just an easier one to fix because we do have a standard environment variable $DESI_ROOT=/global/cfs/cdirs/desi already set.